core/src/prediction/LocalLinearPredictionStrategy.cpp

```diff
@@ -28,12 +28,12 @@
 const size_t LocalLinearPredictionStrategy::OUTCOME = 0;
 
 size_t LocalLinearPredictionStrategy::prediction_length() {
-  return 1;
+  return lambda.size();
 }
 
 LocalLinearPredictionStrategy::LocalLinearPredictionStrategy(const Data* original_data,
                                                              const Data* test_data,
-                                                             double lambda,
+                                                             std::vector<double> lambda,
```

Comment:
(nitpick) I think it'd be clearer to call this `lambdas`, and refer to `lambda` below instead of `lambda_iter`.